### PR TITLE
Update CONTRIBUTING, diagrams for PRs, PivotalTracker

### PR DESCRIPTION
**Pull request by weedySeaDragon**
PT Story:  https://www.pivotaltracker.com/story/show/150721164

### Changes proposed in this pull request:
1. updated the diagram that shows how we use PivotalTracker, including when we "press the button" in PivotalTracker to change the state of the PT task
2. updated the diagram that shows the basic process for doing a PR, including the major branches (e.g. `master` `develop` and the branch for the specific task)
3. updated CONTRIBUTING.md so that it also reflects our current PT process, and our PR branch naming convention
4. moved files into `docs/dev-workflow` (.puml files and the style file associated wit it) and deleted docs not actually used

Please  check the diagrams to make sure that I have them right!

Ready for review:
@AgileVentures/shf-project-team

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -97,12 +99,12 @@ Here is an overview of the general process for contributing (working on a new fe
...
 We use [Pivotal Tracker](https://www.pivotaltracker.com/n/projects/1904891) to manage our work on features, chores and bugfixes.
   
-Work is defined in Pivotal Tracker.  Discussions take place there, in Slack, and in live conversations in scrums and pair programming (currently uses Google Hangouts). 
+Work is defined in Pivotal Tracker.  Discussions take place there, in Slack, and in live conversations in scrums and pair programming (currently we mostly use Google Hangouts). 
 
 ### Features
 Any feature should include Cucumber acceptance tests and RSpec tests where appropriate.
 
-We try to avoid view and controller specs, and focus purely on unit tests at the model and service level where possible.  
+We try to avoid view and controller specs, and focus purely on unit tests at the model and service level where possible.  However, there are times when the functionality is complex enough in a controller that it merits a unit test (RSpec).  Views should be thoroughly tested and exercised with feature (cucumber) tests.
```
Comment (patmbolger) [inline CONTRIBUTING.md]: "Generally we do not use unit-level tests for views.  But we do unit tests for situations that require those - for instance, 1) confirming that menu views are rendered consistent with our menu structure design, and 2) confirming the presence of hidden elements, HTML attributes, etc. that could not be easily tested (if at all) in a cucumber test."

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -189,14 +194,22 @@ git pull upstream develop
 
 Be sure to create your PR against the **develop** branch!
 
-Put a link to the PivotTracker story this addresses so that we keep the connection between the two systems (PivotalTracker and GitHub).  
+We connect the PR in GitHub to the PivotalTracker story manually (simple, but quick and explicit):
+- put a link to the PivotalTracker task in the PR description, _and_
```
Comment (patmbolger) [inline CONTRIBUTING.md]: We refer to "story" in the line above, and now switch to "task".   Suggest we use "story" throughout.

**Comment by patmbolger** [comment]
PT Diagram:

1. "Deployed to heroku ...." button"
  - enter state action: click _finish_, _deliver_ buttons in meeting"
  - exit state action: development branch changes merged to heroku

2. "Finished waiting for scrum review" - perhaps rename to "Finished scrum review"

3. Add button "Deployed to Heroku" - this is for stories that are not client-facing.  It would be at the same level as "Deployed to heroku and waiting for client review".  This would be the end point for the state transition "approved_in_scrum[:is_not_client_facing]"
   this state transitions to "Deployed to production"

4.  "Deployed to production" button -
  - enter state action: merge changes from Heroku to Digital Ocean

NOTE: My changes are trying to reinforce 1) we move all pending-production items to our staging area (heroku) before the client meeting, and 2) we need an explicit action involving moving changes from heroku to DO.

**Review by patmbolger — COMMENTED** [review]
Please see comments.

File: CONTRIBUTING.md (hunk 1 above)
Comment (weedySeaDragon) [reply CONTRIBUTING.md]: I've incorporated this.  I did keep the sentence about `Views should be thoroughly tested...`

**Comment by RobertCram** [comment]
I couldn't easily put my proposed changes in words, so I made a quick drawing (I left out the enter states, etc. They are supposed to be the same is in your state diagram)

untitled_diagram_xml_-_draw_io

**Comment by weedySeaDragon** [comment]
@RobertCram Let me know if I've captured things OK.

**Comment by RobertCram** [comment]
Looks great Ashley!

**Review by RobertCram — APPROVED** [review]

**Review by patmbolger — APPROVED** [review]

**Pull request merged**